incubator/013-signalstop/demo-signal.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+function trap_sigint { echo "Exiting by SIGINT";  exit 0; }
+function trap_sigterm { echo "Exiting by SIGTERM"; exit 0; }
+function trap_sigusr1 { echo "Exiting by SIGUSR1"; exit 0; }
+function trap_sigrtmin3 { echo "Exiting by SIGRTMIN+3"; exit 0; }
```

Comment:
```suggestion
function trap_signal {
  local signal="$1"
  echo -e "\nExiting by ${signal}" >&2
  exit 0
}

for signal in SIGINT SIGTERM SIGUSR1 "SIGRTMIN+3"
do
  trap "trap_signal '${signal}'" "${signal}"
done
```

I like the suggestion and I add a bit more for removing code duplication that I included initially.